src/mixins/itext_behavior.mixin.js

```diff
@@ -562,26 +560,21 @@
       var canDrop = !e.defaultPrevented && this.canDrop(e);
       if (!this.__isDraggingOver && canDrop) {
         this.__isDraggingOver = true;
-        this.enterEditing(e);
-        this.__isDragging && this.abortCursorAnimation();
       }
       else if (this.__isDraggingOver && !canDrop) {
         //  drop state has changed
         this.__isDraggingOver = false;
         !this.__isDragging && this.clearContextTop();
-        this.exitEditing();
       }
-      if (this.__isDraggingOver) {
+      if (this.__isDraggingOver && canDrop) {
         //  can be dropped, inform browser
         e.preventDefault();
         //  inform event subscribers
         options.canDrop = true;
         options.dropTarget = this;
-        //  render
-        this.setCursorByClick(e);
-        this._updateTextarea();
-        this.restartCursorIfNeeded();
-        this.renderCursorOrSelection();
+        // find cursor under the drag part.
+        var dragSelection = this.getSelectionStartFromPointer(e);
+        this.renderCursorAt(dragSelection);
```

Comment:
If I understand correctly we should render the selection here in case `dragSource===dragOverTarget`

Reply:
https://github.com/fabricjs/fabric.js/blob/2db76ec809ef3daf33c37e44f40f23ddb038fb42/src/shapes/itext.class.js#L335

Reply:
In the pr description i wrote that:
```
One regression for sure is there, and is the fact that if i drag/drop in the same text box the dragStart selection isn't visible.
```

I know why this is broken, and is because of the double clear textarea, i was supposed to fix it yesterday but then i couldn't.